### PR TITLE
Support for //link//light

### PR DESCRIPTION
**Pull request by ahcorde**
# 🎉 New feature

## Summary

it requires https://github.com/gazebosim/sdformat/pull/989

Related with this issue https://github.com/gazebosim/gz-mujoco/issues/11

This PR adds lights to links

Simulate : sphere_071

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by ahcorde** [comment]
it requires https://github.com/gazebosim/sdformat/pull/1013

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,12 +27,13 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y libsdformat13-dev python3-ignition-math7 python3-sdformat13
     - name: Install test dependencies
-      run: python3 -m pip install -U tox
+      run: python3 -m pip install -U tox build[virtualenv]
```
Comment (azeey) [inline .github/workflows/ci.yml:30]: This is not needed anymore

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,12 +27,13 @@ jobs:
...
     - name: Test
       env:
         PYTHONPATH: ${{ env.pythonLocation }}/lib/python${{ matrix.python }}/site-packages:/usr/lib/python3/dist-packages
       run: |
         cd sdformat_mjcf
+        python3 -c "import sys;print(sys.path)"
```
Comment (azeey) [inline .github/workflows/ci.yml:36]: remove

File: sdformat_mjcf/sdformat_mjcf/converters/light.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    type = light.type()
+    light = body.add("light",
+                     name=light.name(),
+                     pos=su.vec3d_to_list(pose.pos()),
+                     directional=sdf.Light.LightType.DIRECTIONAL == type,
+                     castshadow=light.cast_shadows(),
+                     attenuation=[light.constant_attenuation_factor(),
+                                  light.linear_attenuation_factor(),
+                                  light.quadratic_attenuation_factor()],
+                     dir=su.vec3d_to_list(light.direction()),
```
Comment (azeey) [inline sdformat_mjcf/sdformat_mjcf/converters/light.py:44]: Do we need to take into account the rotation in `pose` when setting the direction? Otherwise, this direction will be in world frame?

File: sdformat_mjcf/tox.ini [hunk 4]
```diff
@@ -2,9 +2,9 @@
 # for the original commented file
 
 [tox]
-envlist = py{37,38,39,310}
+envlist = py37,py38,py39,py310
```
Comment (azeey) [inline sdformat_mjcf/tox.ini:5]: Are these changes necessary?

**Review by azeey — COMMENTED** [review]
Looks good. My main question is about the direction of the light and how it interacts with the light's pose

File: sdformat_mjcf/sdformat_mjcf/converters/light.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+from ignition.math import Quaterniond
+
+import math
+
+import sdformat_mjcf.sdf_utils as su
+import sdformat as sdf
+
+
+def add_light(body, light, pose,
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply sdformat_mjcf/sdformat_mjcf/converters/light.py]: https://github.com/gazebosim/gz-mujoco/pull/19/commits/33609d745967174b81b2fc0ac927292f9b1756bd

File: sdformat_mjcf/tox.ini [hunk 6]
```diff
@@ -4,7 +4,7 @@
 [tox]
 envlist = py{37,38,39,310}
 
-minversion = 3.3.0
+minversion = 3.7.0
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/tox.ini:7]: I see this was resolved, but I didn't see a change or an explanation.

[the remark this answers is not in the document]
Comment (ahcorde) [reply sdformat_mjcf/tox.ini:7]: sorry, as we are testing from 3.7, why are we setting this to 3.3?

[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/tox.ini:7]: I don't know. @j-rivero can this be 3.7?

[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/tox.ini:7]: Won't block on this. @j-rivero, when you get a chance, feel free to change this back to 3.3 if it needs to be 3.3.

**Review by azeey — APPROVED** [review]
